packages/mobile/src/web3/dataEncryptionKey.ts

```diff
@@ -44,8 +54,12 @@ export function* doFetchDataEncryptionKey(address: string) {
     contractKit.contracts,
     contractKit.contracts.getAccounts,
   ])
-  const dek: string = yield call(accountsWrapper.getDataEncryptionKey, address)
-  yield put(updateAddressDekMap(address, dek || null))
+  const walletToAccountAddress: WalletToAccountAddressType = yield select(
+    walletToAccountAddressSelector
+  )
+  const accountAddress = walletToAccountAddress[walletAddress] || walletAddress
```

Comment:
It may not be wise to default to `walletAddress` if the key is not found, because that indicates that mapping is not set, whereas if it is correctly set to the EOA the mapping will return the `walletAddress`. The default may need to be a fetch on the account synchronously instead. (Also, the `??` operator is more appropriate here than the `||` operator)

Reply:
> It may not be wise to default to `walletAddress` if the key is not found, because that indicates that mapping is not set, whereas if it is correctly set to the EOA the mapping will return the `walletAddress`

I don't think that's necessarily true. The vast majority of users today have their DEK's registered and both their `accountAddress` and `walletAddress` set as their EOA address on the Accounts contract. That's why it seemed reasonable to me to default to checking the user's EOA if an `accountAddress` didn't exist in their locally stored mapping.

Ultimately, I don't see much downside to doing this because if a DEK is not found, the comment just won't be encrypted. The worst case scenario is that comments get encrypted with a DEK the user no longer owns. I don't think we have to deal with that until users can recover their account without their EOA (because the DEK is derived from their EOA mnemonic).

> The default may need to be a fetch on the account synchronously instead
Can you please expand on this?

Reply:
Ah, I see your point. LGTM 👍